src/pymor/models/basic.py

```diff
@@ -84,6 +86,48 @@ def __str__(self):
     def _compute_solution(self, mu=None, **kwargs):
         return self.operator.apply_inverse(self.rhs.as_range_array(mu), mu=mu)
 
+    def _compute_solution_d_mu_single_direction(self, parameter, index, solution, mu):
+        lhs_d_mu = self.operator.d_mu(parameter, index).apply(solution, mu=mu)
+        rhs_d_mu = self.rhs.d_mu(parameter, index).as_range_array(mu)
+        rhs_operator = rhs_d_mu - lhs_d_mu
+        return self.operator.apply_inverse(rhs_operator, mu=mu)
+
+    _compute_allowed_kwargs = frozenset({'use_adjoint'})
+
+    def _compute_output_d_mu(self, solution, mu, use_adjoint=True):
+        """compute the gradient of the output functional  w.r.t. the parameters
+
+        Parameters
+        ----------
+        solution
+            Internal model state for the given |Parameter value|
+        mu
+            |Parameter value| for which to compute the gradient
+        use_adjoint
+            Use the adjoint solution for a more efficient way of computing the gradient.
+            See Section 1.6.2 in [HPUU09]_ for more details.
+            So far, this approach is only valid for linear output functionals and symmetric operators.
```

Comment:
```suggestion
            So far, this approach is only valid for linear models.
```

(For nonlinear operators, the adjoint isn't defined.)